### ISSUE TITLE
Resolving LinkageError for iP submission

### ISSUE THREAD

**Issue by chrus-chong** [nus-cs2103-AY2122S1/forum#264]
I am working on Microsoft Windows 10 Home. Prior to submission of the iP, I created a JAR file using shadowJar.
I made sure to test it out myself using Java 11 (as seen below) and sent it to two friends who said that it worked perfectly on their laptops(one mac user and one windows user).

![image](https://user-images.githubusercontent.com/85006125/134136712-7644da47-b36d-4d8a-aafb-f31482d2647f.png)

However, today I received an email from the CS2103 Teaching Team of LinkageErrors plaguing my JAR files upon running.

On Windows, using Java 11, for the Duke.Productivity.Tool.jar uploaded on 09-17-2021 (MM-DD-YYYY) at 14:13:14:
```Error: LinkageError occurred while loading main class duke.Launcher java.lang.UnsupportedClassVersionError: duke/Launcher has been compiled by a more recent version of the Java Runtime (class file version 57.0), this version of the Java Runtime only recognizes class file versions up to 55.0```
On Linux, using Java 11, for the Duke.Productivity.Tool.jar uploaded on 09-17-2021 (MM-DD-YYYY) at 14:13:14:
```Error: LinkageError occurred while loading main class duke.Launcher java.lang.UnsupportedClassVersionError: duke/Launcher has been compiled by a more recent version of the Java Runtime (class file version 57.0), this version of the Java Runtime only recognizes class file versions up to 55.0```
On Mac, using Java 11, for the Duke.Productivity.Tool.jar uploaded on 09-17-2021 (MM-DD-YYYY) at 14:13:14:
```Error: LinkageError occurred while loading main class duke.Launcher java.lang.UnsupportedClassVersionError: duke/Launcher has been compiled by a more recent version of the Java Runtime (class file version 57.0), this version of the Java Runtime only recognizes class file versions up to 55.0```

Upon closer inspection, my JAR file and Gradle runs the application without any issues, however, when I hit the play button in my Launcher class, they give me the errors as stated above. This is odd as my Build.Gradle file specifies that the Launcher class be executed when I do the run task. My launcher class comprises of a mere 3 lines and given that everything else works, I am unsure of how to resolve this issue. Any help is greatly appreciated! Thank you very much!

```
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Noticed that @jianoway faced this issue before but managed to solve it on his/her own. Any help is greatly appreciated! Thank you very much!

**Comment by damithc** [comment]
Some common remedies you can try:
* Check these settings
  ![image](https://user-images.githubusercontent.com/1673303/133767453-b2166add-499f-4829-8788-c3ae8a4186c7.png)
  If the above settings are fine, run gradle using intellij UI rather than the terminal
* Set JAVA_HOME system variable to point to JDK 11 folder
* Remove newer JVM/JDK versions (if not needed for other modules)

**Comment by chrus-chong** [comment]
Thank you very much for the help Prof @damithc! For anyone who's facing similar issues in the future, setting the JAVA_HOME variable to point to the JDK 11 folder as prof suggested and Issue [#226](https://github.com/nus-cs2103-AY2122S1/forum/issues/226) helped me to resolve this issue. I ran the clean task as well as shadowJar once more to create a new Jar file.

As the smoke tests are conducted only once per student prior to submission, may I trouble you to help me run the attached Jar file and see if all is well?
[Duke Productivity Tool.zip](https://github.com/nus-cs2103-AY2122S1/forum/files/7207638/Duke.Productivity.Tool.zip)

**Comment by damithc** [comment]
> As the smoke tests are conducted only once per student prior to submission, may I trouble you to help me run the attached Jar file and see if all is well?

Any classmates want to help @chrus-chong with this? i.e., confirm the JAR file can launch normally with JDK 11

**Comment by rohit0718** [comment]
works on my com chrus! looks like all good alr @chrus-chong

**Comment by chrus-chong** [comment]
Thank you very much for all the help and assistance!
